### PR TITLE
Development : Created the Added Loan Feature 

### PR DESCRIPTION
**Pull request by omidiora**
I created a  add loan feature in which the authenticated user can create a loan .. it uses token as the user id to verfiy the user . the models.py
 contains  Amount, due_date, interest_rate, borrower/lender, description. 
The foreign key is attach to to the user.

the database fields contain: user as (foreignkey), date_created, due_date, interest_rate, paid(boolean), borrower/lender,  description, and balance_to_pay

it return response 201 if successfully  created.

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -14,3 +14,22 @@ def __str__(self):
...
+
+class add_loan_record(models.Model):
+    loan=models.CharField(max_length=200, null=True)
+    user_id=models.ForeignKey(User,on_delete=models.CASCADE)
+    due_date=models.DateField(auto_now=True)
+    created=models.DateField(auto_now_add=True)
+    amount=models.TextField()
+    interest_rate=models.IntegerField()
+    paid=models.BooleanField(default=False)
+    borrower=models.CharField(max_length=200)
```
Comment (nextwebb) [inline api/models.py]: The borrower field should be a Boolean, because a borrower as well a lender can create loan records.
This field should be a Boolean.

File: api/models.py [hunk 2]
```diff
@@ -14,3 +14,22 @@ def __str__(self):
...
+
+
+
+class add_loan_record(models.Model):
+    loan=models.CharField(max_length=200, null=True)
+    user_id=models.ForeignKey(User,on_delete=models.CASCADE)
+    due_date=models.DateField(auto_now=True)
+    created=models.DateField(auto_now_add=True)
+    amount=models.TextField()
+    interest_rate=models.IntegerField()
```
Comment (nextwebb) [inline api/models.py]: The Interest field should accept blank and data type should be characters.

File: api/models.py [hunk 3]
```diff
@@ -14,3 +14,22 @@ def __str__(self):
...
+
+
+
+
+class add_loan_record(models.Model):
+    loan=models.CharField(max_length=200, null=True)
+    user_id=models.ForeignKey(User,on_delete=models.CASCADE)
+    due_date=models.DateField(auto_now=True)
+    created=models.DateField(auto_now_add=True)
+    amount=models.TextField()
```
Comment (nextwebb) [inline api/models.py]: Amount should integers or characters

File: api/models.py [hunk 4]
```diff
@@ -14,3 +14,22 @@ def __str__(self):
...
+    interest_rate=models.IntegerField()
+    paid=models.BooleanField(default=False)
+    borrower=models.CharField(max_length=200)
+    description=models.TextField(blank=True)
```
Comment (nextwebb) [inline api/models.py]: Why is description default set to blank?

File: api/models.py [hunk 5]
```diff
@@ -14,3 +14,22 @@ def __str__(self):
...
         return f'{self.user.first_name}'
 
+
+
+
+
+class add_loan_record(models.Model):
+    loan=models.CharField(max_length=200, null=True)
+    user_id=models.ForeignKey(User,on_delete=models.CASCADE)
+    due_date=models.DateField(auto_now=True)
```
Comment (nextwebb) [inline api/models.py]: Duedate should not be auto_now. It should be specifically set by the user

File: api/models.py [hunk 6]
```diff
@@ -14,3 +14,22 @@ def __str__(self):
         """one-line docstring for representing the Profile object."""
         return f'{self.user.first_name}'
 
+
+
+
+
+class add_loan_record(models.Model):
+    loan=models.CharField(max_length=200, null=True)
```
Comment (nextwebb) [inline api/models.py]: What's this loan field doing? I don't understand

File: api/serializers.py [hunk 7]
```diff
@@ -38,3 +38,13 @@ class Meta:  #pylint: disable=too-few-public-methods
...
         fields = ('id', 'facebook_user', 'phone', 'profile',)
+
+
+
+class AddLoanSerializer(serializers.ModelSerializer):
+    #Adding of load detail
+    class Meta:
+        #return option Loan
+            model=add_loan_record
+            fields=('loan','user_id','created','amount','interest_rate','paid','borrower','description','balance_to_pay')
```
Comment (nextwebb) [inline api/serializers.py]: Most of your fields were set to blank True, so you might get errors here because they were never parsed

File: api/urls.py [hunk 8]
```diff
@@ -1,11 +1,12 @@
 from django.urls import path
 from rest_framework import routers
 from django.conf.urls import include
-from .views import UserViewSet, ProfileViewSet
+from .views import UserViewSet, ProfileViewSet,AddLoanViewSet
 
 router = routers.DefaultRouter()
 router.register('user', UserViewSet)
 router.register('profile', ProfileViewSet)
+router.register('loan/add', AddLoanViewSet)
```
Comment (nextwebb) [inline api/urls.py:9]: Did you test this route?

File: requirements.txt [hunk 9]
```diff
@@ -1,7 +1,96 @@
...
+django-celery-results==1.2.1
+django-cors-headers==3.3.0
+django-crispy==1.8.1
+django-crispy-forms==1.9.0
+django-dotenv==1.4.2
+django-filter==2.2.0
+django-rest-auth==0.9.5
+django-rest-swagger==2.0.7
 djangorestframework==3.11.0
-Pillow==7.2.0
```
Comment (nextwebb) [inline requirements.txt:4]: Why did you remove pillow package?

**Review by nextwebb — COMMENTED** [review]
Didn't meet the acceptance criteria

File: api/models.py (hunk 2 above)
Comment (omidiora) [reply api/models.py]: the interest rate would be in number(figure) an that is why is in integer....... I would change it to DecimalField because interest rate can be in decimal fiqure

File: api/models.py (hunk 4 above)
Comment (omidiora) [reply api/models.py]: i have remove the default (blank)

File: api/models.py (hunk 5 above)
Comment (omidiora) [reply api/models.py]: 0k

File: api/models.py (hunk 6 above)
Comment (omidiora) [reply api/models.py]: this  is the amount of loan that the users(borrower and lender)  will be agreeing on that it what it meant for

File: api/serializers.py (hunk 7 above)
Comment (omidiora) [reply api/serializers.py]: not all except for **balance_to_pay,  description ,Loan** 
which i have remove the  descrition(blank=True) in the new commit 


carbon

File: api/urls.py (hunk 8 above)
Comment (omidiora) [reply api/urls.py:9]: yes I  did , that is the route for  creating loan

File: requirements.txt (hunk 9 above)
Comment (omidiora) [reply requirements.txt:4]: not intentional ,, that was when i updated my requirement.txt

**Comment by nextwebb** [comment]
bro your PR didnt pass the codacy test

**Comment by nextwebb** [comment]
here's also the acceptance criteria for your PR. pls check it out
https://wuralab.atlassian.net/browse/DNS-12?atlOrigin=eyJpIjoiYjBhZjZmNmRmYTFiNDNhNjg0ZTMyZjQ5MTcxN2VjYjEiLCJwIjoiaiJ9

**Comment by nextwebb** [comment]
hurray ! @omidiora PR finalled passed all them tests

**Comment by nextwebb** [comment]
i trust its healthy PR @omidiora

**Review by nextwebb — APPROVED** [review]
there's nothing broken we cant fix @omidiora :)

**Review by simplytunde — APPROVED** [review]

**Comment by simplytunde** [comment]
All clear!
